hare/haretypes.go

```diff
@@ -53,6 +52,21 @@ func (mType MessageType) String() string {
 	}
 }
 
+func (id InstanceId) Id() uint32 {
+	return id.Uint32()
+}
+
+func (id InstanceId) Uint32() uint32 {
```

Comment:
why is this function needed???

Reply:
Go is a (very) strictly typed lang. Even when you declare type MyType int and conversion to int is trivial and safe - it is not done automatically. Hence, I find it neater to have a conversion function. If you ever change int to int64 you will be able to fix all the code accordingly using refactoring (as opposed to explicit conversions).

Reply:
loose this, cast when needed. should not be needed a lot

Reply:
Fixed